probe-rs/src/core/memory_mapped_registers.rs

```diff
@@ -0,0 +1,99 @@
+/// A memory mapped register, for instance ARM debug registers (DHCSR, etc).
+pub trait MemoryMappedRegister<T>: Clone + From<T> + Into<T> + Sized + std::fmt::Debug {
+    /// The register's address in the target memory.
+    /// For some architectures (e.g. ARM Cortex-A) this address is offset from a base address.
+    /// For others (e.g. ARM Cortex-M, RISC-V) this address is absolute.
+    const ADDRESS_OFFSET: u64;
+    /// The register's name.
+    const NAME: &'static str;
+    /// Get the register's address in the memory map.
+    /// For some architectures (e.g. ARM Cortex-A) this address is offset from a base address.
+    /// For others (e.g. ARM Cortex-M, RISC-V) this address is absolute.
+    fn get_mmio_address(base_address: Option<u64>) -> u64 {
+        if let Some(base_address) = base_address {
+            if let Some(mmio_address) = base_address.checked_add(Self::ADDRESS_OFFSET) {
+                mmio_address
+            } else {
+                tracing::error!(
```

Comment:
This should be a panic or error, because right now we will read from a wrong address which cannot be correct.

Reply:
I would prefer to NOT panic, because that has very negative UX for probe-rs-debugger. 

I will modify the function to return an error.